### ISSUE TITLE
import-x/order mismatches typescript language server organize imports for `#` prefixed imports

### ISSUE THREAD

**Issue by Shinigami92** [un-ts/eslint-plugin-import-x#286]
Minimal reproducible: https://github.com/Shinigami92/eslint-plugin-import-x-repro

#### input

```ts
import { internA } from "#a";
import { scopeA } from "@a/a";
import { localA } from "./a";

console.log({ internA, scopeA, localA });
```

#### actual

`import-x/order` reports error

#### expected

should not report, because when running typescript's lsp organize imports, then the above input order is enforced

#### context

the `#` prefix comes from `package.json` `"imports": { ... }` field: https://nodejs.org/api/packages.html#imports

**Comment by JounQin** [comment]
It depends on where the final module path resolved, mostly, `#a` would be resolved as internal group, `imports` as top is not a already-have feature, but I think it could be add a new option for this?

**Comment by Shinigami92** [comment]
> It depends on where the final module path resolved, mostly, `#a` would be resolved as internal group, `imports` as top is not a already-have feature, but I think it could be add a new option for this?

Hey @JounQin 🙂 
I personally do not care how it is implemented under the hood, either with or without a new group. But the default behavior should be compatible with TS LSP.

I'm also not sure if this is an issue in `eslint-plugin-import-x` or a dependency the plugin is just using.

But I'm potentially willing to implement the fix with a PR if I get some instructions where to start. 😺

**Comment by Shinigami92** [comment]
> There is no such convention?

I'm not sure if there is something like it in `eslint-plugin-import-x`

In @simonhaenisch's `prettier-plugin-organize-imports` we called the native language server function: https://github.com/simonhaenisch/prettier-plugin-organize-imports/blob/e79ed64e5ecd05bdd65140b814b37c9a1a360767/lib/organize.js#L24-L30

This is then the same as calling it in VSCode manually: https://code.visualstudio.com/docs/typescript/typescript-refactoring#_organize-imports

> I'm fine with adding a new option, but I don't know how to name it, you can give your choice via a PR. The sorting ranks happen at
> 
> [eslint-plugin-import-x/src/rules/order.ts](https://github.com/un-ts/eslint-plugin-import-x/blob/877bbbb4d7795e2adc9d32e99073f516d02fe815/src/rules/order.ts#L715)
> 
> Line 715 in [877bbbb](/un-ts/eslint-plugin-import-x/commit/877bbbb4d7795e2adc9d32e99073f516d02fe815)
> 
>  function computeRank(

OK thx, I will have a look into it after company work in my private time 👍

**Comment by JounQin** [comment]
> I'm not sure if there is something like it in eslint-plugin-import-x

I know those tools follow TS LSP by using it under the hood, but `eslint-plugin-import(-x)` and `eslint-plugin-simple-import-sort` are created before such things AFAIK, we could try to **match** what you proposed but not using LSP service which requires `typescript` in dependency.

**Comment by JounQin** [comment]
`typescript` is surely already a dev dep, it's pure TypeScript project. 🍺
